packages/core/test/request-builder/v4/get-all-request-builder.spec.ts

```diff
@@ -72,6 +80,58 @@ describe('GetAllRequestBuilderV4', () => {
       ]);
     });
 
+    it('type tests', async () => {
+      // TestEntityTemporal<string, number>[]
+      const res1 = await TestEntityTemporal.requestBuilder()
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, number>[]
+      const res2 = await TestEntityTemporal.requestBuilder(defaultMiddleware)
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, string>[]
+      const res3 = await TestEntityTemporal.requestBuilder(customMiddlewareFull)
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, string>[]
+      const res4 = await TestEntityTemporal.requestBuilder(
+        customMiddlewarePartial
+      )
+        .getAll()
+        .execute(defaultDestination);
```

Comment:
The examples above illustrate how the `requestBuilder()` affects the return type.